chain/client/src/test_utils.rs

```diff
@@ -903,9 +903,14 @@ pub fn setup_mock_all_validators(
                             if aa.get(&key).is_none() {
                                 aa.insert(key);
                                 for (_, view_client) in connectors1.read().unwrap().iter() {
-                                    view_client.do_send(NetworkViewClientMessages::AnnounceAccount(
-                                        vec![(announce_account.clone(), None)],
-                                    ))
+                                    let view_client = view_client.clone();
+                                    let announce_account = announce_account.clone();
+                                    let fut = actix::fut::wrap_future(async move {
+                                        let _ = view_client.send(NetworkViewClientMessages::AnnounceAccount(
+                                            vec![(announce_account, None)],
+                                        )).await;
+                                    });
+                                    ctx.spawn(fut);
```

Comment:
it is critical to replace a.do_send with ctx.spawn(wrap_future(a.send())). Perhaps having a function which does exactly that: 
fn new_do_send(ctx:Context,f:Future) { ctx.spawn(wrap_future(f)) }
would make it more straightforward to do the migration, at the same time putting an emphasis on the semantical difference between send and do_send?

Also note that send may fail if the mailbox is full, while do_send always succeeds. IMO we shouldn't fail silently in such a case.

Reply:
This is the only case in the codebase where we use do_send for view client, and it feels like this is no-op basically, so I'd rather refactor this away completely. But for that I need to understand how the stuff works sadly :)

Reply:
ack, also this is testonly code, so it doesn't matter I guess.
Given that this is testonly though I'd make it panic on error - at worst some test will get flaky.